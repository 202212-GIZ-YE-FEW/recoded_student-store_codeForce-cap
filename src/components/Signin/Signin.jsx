import { withTranslation } from "next-i18next"
import Image from "next/image"
import Link from "next/link"
import { useRouter } from "next/router"
import { useState } from "react"
import { BsFacebook, BsGoogle, BsTwitter } from "react-icons/bs"
import { ToastContainer, toast } from "react-toastify"

import "react-toastify/dist/ReactToastify.css"
import styles from "./Signin.module.css"

import signIn from "@/utils/firebase/signin"
import { signinValidation } from "@/utils/schemaValidations/signin"

import Button from "../button"
import Input from "../input"

function Signin({ t }) {
  const router = useRouter()

  const [formData, setFormData] = useState({
    email: "",
    password: "",
  })

  const [errors, setErrors] = useState({})

  const handleChange = (event) => {
    const { name, value } = event.target
    setFormData((prevState) => ({
      ...prevState,
      [name]: value,
    }))
  }

  const handleSubmit = async (event) => {
    event.preventDefault()

    try {
      await signinValidation.validate(formData, { abortEarly: false })
      toast.warn("Please wait")
      const { email, password } = formData

      const { error } = await signIn(email, password)

      if (error) {
        return toast.error(error)
      }

      // else when successful
      return router.replace("/").then(() => {
        toast.success("Welcome back")
      })
    } catch (err) {
      const validationErrors = {}
      err.inner.forEach((error) => {
        validationErrors[error.path] = error.message
      })
      setErrors(validationErrors)
    }
  }

  return (
    <>
      <ToastContainer pauseOnHover={false} newestOnTop={true} />
      <div className={`flex justify-center  md:flex-row  bg-[#f1f6fa] `}>
        <div className={` ${styles.handbox_background}   w-3/5 `}>
          <div className=' p-20 '>
            <Image
              src='/images/hands_box.png'
              alt='handbox'
              width={500}
              height={500}
            />
          </div>
        </div>
        <div className='container m-auto flex w-5/6 flex-col items-center'>
          <h1 className='my-2 py-6 text-4xl font-semibold text-[#FF8A57] md:my-3 md:text-5xl'>
            {t("sign-in")}
          </h1>
          <form
            onSubmit={handleSubmit}
            className='container m-auto mb-6 flex w-5/6 flex-col items-center'
          >
            <label htmlFor='email'>
              <Input
                type='text'
                name='email'
                placeholder={t("email")}
                className='lg:w-96 md:w-72'
                value={formData.email}
                onChange={handleChange}
              />
            </label>
            {toast.error(errors.email).email}
            <label htmlFor='password'>
              <Input
                type='password'
                name='password'
                placeholder={t("password")}
                className='lg:w-96 md:w-64'
                value={formData.password}
                onChange={handleChange}
              />
            </label>
            {toast.error(errors.password).password}
            <div className='my-4 gap-1 flex-row flex lg:justify-center'>
              <Button
                className='orangeSignIn'
                text={t("sign-in")}
                type='submit'
              />
              <Button className='forgotPassword' text={t("forgot-password")} />
            </div>
          </form>
          <div className='flex items-center'>
            <div className='my-1 mr-2 h-px mt-[10px] w-[164px] bg-[#9dafbd]'></div>
            <p>{t("or")}</p>
            <div className='my-1 mr-2 h-px mt-[10px] w-[164px] bg-[#9dafbd]'></div>
          </div>
          <p className='text-md m-1 text-[#647581]'>{t("sign-in-with")}</p>
          <div className='m-1 mb-8 flex flex-row  '>
            <button className=' m-1 flex items-center rounded-3xl border border-[#F26F6F] p-1  text-[#F26F6F]'>
              <BsGoogle color='#F26F6F' size={24} style={{ padding: "1px" }} />
              <p className='mx-2 text-sm md:mx-3'>{t("google")}</p>
            </button>
            <button className='color-darkPurple m-1  flex items-center rounded-3xl border border-[#485DCF] p-1 text-[#485DCF]'>
              <BsFacebook
                color='#485DCF'
                size={24}
                style={{ padding: "1px" }}
              />
              <p className='mx-2 text-sm md:mx-3'>{t("facebook")}</p>
            </button>
            <button className=' m-1 flex justify-around rounded-3xl border border-[#28C7FA]  p-1 text-[#28C7FA] '>
              <BsTwitter color='#28C7FA' size={24} style={{ padding: "1px" }} />
              <p className='mx-2 text-sm md:mx-3'>{t("twitter")}</p>
            </button>
          </div>
          <div className='mb-4 text-xl text-[#647581]'>
            <p>{t("dont-have-an-account")}</p>
          </div>
          {/* <Button className='orangeSignIn ' text='Sign-up' /> */}
          <Link
            href='/signup'
            className='h-10 w-40 lg:w-48 lg:px-5 p-1 px-4 rounded-lg border-1 bg-pumpkin text-md font-normal text-white shadow-sm text-center'
          >
            {t("sign-up")}
          </Link>
        </div>
      </div>
    </>
  )
}

export default withTranslation("signin")(Signin)
