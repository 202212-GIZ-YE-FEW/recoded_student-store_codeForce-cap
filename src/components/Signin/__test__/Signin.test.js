import renderer from "react-test-renderer"

import Signin from ".."

it("renders correctly", () => {
  const tree = renderer.create(<Signin />).toJSON()
  expect(tree).toMatchSnapshot()
})
