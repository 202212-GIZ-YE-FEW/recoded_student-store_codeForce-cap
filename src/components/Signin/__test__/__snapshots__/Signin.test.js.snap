// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-center  md:flex-row  bg-[#f1f6fa] "
>
  <div
    className=" handbox_background   w-3/5 "
  >
    <div
      className=" p-20 "
    >
      <img
        alt="handbox"
        data-nimg="1"
        decoding="async"
        height={500}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fhands_box.png&w=1080&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fhands_box.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fhands_box.png&w=1080&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={500}
      />
    </div>
  </div>
  <div
    className="container m-auto flex w-5/6 flex-col items-center"
  >
    <h1
      className="my-2 py-6 text-4xl font-semibold text-[#FF8A57] md:my-3 md:text-5xl"
    >
      sign-in
    </h1>
    <form
      className="container m-auto mb-6 flex w-5/6 flex-col items-center"
    >
      <input
        autoComplete="off"
        className="my-2 block w-full rounded-lg border border-gray-300 p-2.5 text-center lg:w-96 md:w-72"
        name="firstName"
        placeholder="name"
        required={false}
        type="text"
      />
      <input
        autoComplete="off"
        className="my-2 block w-full rounded-lg border border-gray-300 p-2.5 text-center lg:w-96 md:w-64"
        name="surname"
        placeholder="surname"
        required={false}
        type="text"
      />
      <div
        className="my-4 gap-1 flex-row flex lg:justify-center"
      >
        <button
          className="h-10 w-40 lg:w-48 lg:px-5 p-1 px-4 rounded-lg border-1 bg-pumpkin text-md font-normal text-white shadow-sm"
          type="button"
        >
          sign-in
        </button>
        <button
          className="h-10 w-40 lg:w-48 lg:px-5 p-1 px-4 rounded-lg border-1 bg-pumpkin text-md font-normal text-white shadow-sm"
          type="button"
        >
          forgot-password
        </button>
      </div>
    </form>
    <div
      className="flex items-center"
    >
      <div
        className="my-1 mr-2 h-px mt-[10px] w-[164px] bg-[#9dafbd]"
      />
      <p>
        or
      </p>
      <div
        className="my-1 mr-2 h-px mt-[10px] w-[164px] bg-[#9dafbd]"
      />
    </div>
    <p
      className="text-md m-1 text-[#647581]"
    >
      sign-in-with
    </p>
    <div
      className="m-1 mb-8 flex flex-row  "
    >
      <button
        className=" m-1 flex items-center rounded-3xl border border-[#F26F6F] p-1  text-[#F26F6F]"
      >
        <svg
          color="#F26F6F"
          fill="currentColor"
          height={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "#F26F6F",
              "padding": "1px",
            }
          }
          viewBox="0 0 16 16"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.545 6.558a9.42 9.42 0 0 1 .139 1.626c0 2.434-.87 4.492-2.384 5.885h.002C11.978 15.292 10.158 16 8 16A8 8 0 1 1 8 0a7.689 7.689 0 0 1 5.352 2.082l-2.284 2.284A4.347 4.347 0 0 0 8 3.166c-2.087 0-3.86 1.408-4.492 3.304a4.792 4.792 0 0 0 0 3.063h.003c.635 1.893 2.405 3.301 4.492 3.301 1.078 0 2.004-.276 2.722-.764h-.003a3.702 3.702 0 0 0 1.599-2.431H8v-3.08h7.545z"
          />
        </svg>
        <p
          className="mx-2 text-sm md:mx-3"
        >
          google
        </p>
      </button>
      <button
        className="color-darkPurple m-1  flex items-center rounded-3xl border border-[#485DCF] p-1 text-[#485DCF]"
      >
        <svg
          color="#485DCF"
          fill="currentColor"
          height={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "#485DCF",
              "padding": "1px",
            }
          }
          viewBox="0 0 16 16"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
          />
        </svg>
        <p
          className="mx-2 text-sm md:mx-3"
        >
          facebook
        </p>
      </button>
      <button
        className=" m-1 flex justify-around rounded-3xl border border-[#28C7FA]  p-1 text-[#28C7FA] "
      >
        <svg
          color="#28C7FA"
          fill="currentColor"
          height={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "#28C7FA",
              "padding": "1px",
            }
          }
          viewBox="0 0 16 16"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.026 15c6.038 0 9.341-5.003 9.341-9.334 0-.14 0-.282-.006-.422A6.685 6.685 0 0 0 16 3.542a6.658 6.658 0 0 1-1.889.518 3.301 3.301 0 0 0 1.447-1.817 6.533 6.533 0 0 1-2.087.793A3.286 3.286 0 0 0 7.875 6.03a9.325 9.325 0 0 1-6.767-3.429 3.289 3.289 0 0 0 1.018 4.382A3.323 3.323 0 0 1 .64 6.575v.045a3.288 3.288 0 0 0 2.632 3.218 3.203 3.203 0 0 1-.865.115 3.23 3.23 0 0 1-.614-.057 3.283 3.283 0 0 0 3.067 2.277A6.588 6.588 0 0 1 .78 13.58a6.32 6.32 0 0 1-.78-.045A9.344 9.344 0 0 0 5.026 15z"
          />
        </svg>
        <p
          className="mx-2 text-sm md:mx-3"
        >
          twitter
        </p>
      </button>
    </div>
    <div
      className="mb-4 text-xl text-[#647581]"
    >
      <p>
        dont-have-an-account
      </p>
    </div>
    <a
      className="h-10 w-40 lg:w-48 lg:px-5 p-1 px-4 rounded-lg border-1 bg-pumpkin text-md font-normal text-white shadow-sm text-center"
      href="/signup"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      sign-up
    </a>
  </div>
</div>
`;
